pkg/lib/authn/sso/oauth_provider.go

```diff
@@ -42,7 +43,12 @@ func (p *OAuthProviderFactory) getActiveOrDemoProvider(alias string) (provider o
 
 	if config.OAuthSSOProviderConfig(providerConfig).IsDisabled() {
 		// TODO(tung): handle demo status
-		err = api.ErrOAuthProviderInactive
+
+		details := apierrors.Details{
+			"OAuthProviderAlias": alias,
+			"OAuthProviderType":  providerConfig.Type(),
+		}
+		err = api.OAuthProviderInactive.NewWithInfo("oauth provider is inactive because of missing credentials", details)
```

Comment:
I think this error construction should be a method of `config.OAuthSSOProviderConfig` Maybe we should even remove `IsDisabled()` and replace it with `GetUsableProviderConfig`

Reply:
Do "usable" includes demo credential? If yes, the demo credential probably come from app dependencies, so we cannot construct it in config.OAuthSSOProviderConfig. And for the same reason, we can only determine if the error should be returned in OAuthProviderFactory.

Reply:
Got that. Let's keep what we have then.